cirq-google/cirq_google/engine/stream_manager.py

```diff
@@ -330,19 +325,17 @@ def _get_retry_request_or_raise(
             return create_program_and_job_request
     elif error.code == Code.PROGRAM_ALREADY_EXISTS:
         if 'create_quantum_program_and_job' in current_request:
-            raise ProgramAlreadyExistsError(
-                current_request.create_quantum_program_and_job.quantum_program.name
-            )
+            # If the program already exists and is created as part of the stream client, the job
+            # should also exist because they are created at the same time.
+            # If the job is missing, a `CreateQuantumJobRequest` will be issued after a
```

Comment:
How does the program get created w/o the job if they're created together in a CreateProgramAndJobRequest? Does a closed stream kill the server-side handling?

Reply:
Clarified a bit in the comment - it's for the unlikely case that the program is created outside `StreamManager`. The logic here doesn't explicitly try to solve this case, but it just happens to do the right thing.